src/powerset.py

```diff
@@ -1,12 +1,25 @@
-"""Module for computing power sets."""
-
-from typing import Any
-
-
-def power(x: list[Any]) -> list[list[Any]]:
-    """
-    Compute the power-set of x.
-
-    Returns the power-set of x as a list of lists.
-    """
-    return []  # FIXME: you need to add some code here.
+"""Module for computing power sets."""
+
+from typing import Any
+
+def power(x: list[Any]) -> list[list[Any]]:
+    """
+    Compute the power-set of x.
+
+    Returns the power-set of x as a list of lists.
+    """
+    powerset = []
+    for i in range(0, 2**len(x)):
+        bit_representation = format(i, "b").zfill(len(x))
+        idx = 0
+        subset = []
+        for j in bit_representation[::-1]:
+            if j == "0":
+                idx += 1
+                continue
+            elif j == "1":
+                subset.append(x[idx])
+                idx += 1
+        powerset.append(subset)

```

Comment:
When you want to keep track of both the index and elements in a sequence, you can use the `enumerate()` function. That way, you don't risk forgetting to increment `idx` in one of the `if`-branches.

```python
for idx, val in enumerate(x):
    print(f"At x[{idx}] you have value {x[idx]} which is {val}")
```

Reply:
Great tip! Thank you Thomas